src/model.cpp

```diff
@@ -475,6 +475,45 @@ bool Model::hasImports() const
     return importsPresent;
 }
 
+void fixComponentUnits(const ModelPtr &model, const ComponentPtr &component)
+{
+    for (size_t v = 0; v < component->variableCount(); ++v) {
+        auto variable = component->variable(v);
+        if (variable->units() != nullptr) {
+            // Find the units in the model and switch out.
+            auto units = model->units(variable->units()->name());
+            if (units != nullptr) {
+                variable->setUnits(units);
```

Comment:
What happens to the original `variable->units()`?

Reply:
It disappears into the ether.  No references ... created inline ... so that if people called this as a `variable->clone()` function, they can still access the units if they want to.  But if it's come from a `model->clone()` call, then they get fixed up properly.

Reply:
Ok.